src/ray/raylet/worker_pool.cc

```diff
@@ -151,19 +151,27 @@ void WorkerPool::SetAgentManager(std::shared_ptr<AgentManager> agent_manager) {
   agent_manager_ = agent_manager;
 }
 
+void WorkerPool::PopWorkerCallbackAsync(const PopWorkerCallback callback,
+                                        std::shared_ptr<WorkerInterface> worker) {
+  if (callback) {
+    io_service_->post([callback, worker]() { callback(worker); });
```

Comment:
Why do we have to post here?

Reply:
I want to make the code execute in order. But I found this bring a lot problem in test case. So I have delete this post.